pkg/kubelet/kubelet_node_status.go

```diff
@@ -91,7 +93,15 @@ func (kl *Kubelet) tryRegisterWithAPIServer(node *v1.Node) bool {
 		return true
 	}
 
-	if !apierrors.IsAlreadyExists(err) {
+	switch {
+	case apierrors.IsAlreadyExists(err):
+		// Node already exists, proceed to reconcile node.
+	case apierrors.IsForbidden(err):
+		// Creating nodes is forbidden, but node may still exist, attempt to get the node.
+		if utilfeature.DefaultFeatureGate.Enabled(features.DisableKubeletRegisterGetOnForbidden) {
+			return false
```

Comment:
It might be easier to check the logs to find the cause of the failure

```suggestion
			klog.ErrorS(err, "Unable to register node with API server, reason is forbidden", "node", klog.KObj(node))
			return false
```